cylc/uiserver/app.py

```diff
@@ -522,3 +528,23 @@ async def stop_extension(self):
         self.data_store_mgr.executor.shutdown(wait=False)
         # Destroy ZeroMQ context of all sockets
         self.workflows_mgr.context.destroy()
+        self.clean_pid_file()
+
+    @staticmethod
+    def clean_pid_file():
+        """Remove pid file if it exists."""
+        pid_file = Path("~/.cylc/uiserver/pid").expanduser()
+        pid = os.getpid()
+        with contextlib.suppress(
+                FileNotFoundError, json.decoder.JSONDecodeError, KeyError):
+            with open(pid_file, "r") as f:
+                content = json.load(f)
+                del content[str(pid)]
+                json_content = json.dumps(content)
+            with open(pid_file, "w") as f:
+                f.write(json_content)
+        with contextlib.suppress(FileNotFoundError):
+            file_size = os.path.getsize(pid_file)
+            if file_size in [0, 2]:  # delete empty pid file
+                pid_file.unlink()
+                print(f"Removing Process Id file from {pid_file.parent}")
```

Comment:
```suggestion
                print(f"Removed Process ID file from {pid_file.parent}")
```
(Or perhaps change the line order :grin: )